src/test/testdata/lsl/generics.lsl

```diff
@@ -108,11 +108,21 @@ automaton A
 
         }
         var obj3: HashMap<in Int, out string> = x as HashMap<in Int, out string>;
+        
+        var newHashMapWithLiterals: HashMap<in Int, out String> = new A<5, "anyString">(state = Initialized);
+        A(newHashMapWithLiterals)._genericProc<5, 'J', "anyString">();
+        if (x is HashMap<5.79, "anyString">) {
+
+        }
+        var obj4: HashMap<in Int, out string> = x as HashMap<5.79, "anyString">;
     }
 
     fun *.ComplicatedWhere <T, R> (from: R, to: T): void where T: HashMap<HashMap<Int, map<Int, Int>>, Int>, R: out Int {
     }
 
     fun *.genericReturnType <T, R> (): HashMap<HashMap<T, R>, HashMap<T, R>> where T: any, R: any {
     }
+    
+    fun *.genericWhereLiteralTypes <T, R> (): void where T: 5, R: "anyString" {
```

Comment:
As I remember, we have decided to treat `int32`, `string` types for type variable constraints as receiving literals.
Combined with the possibility of allowing multiple versions of the same generic/template declaration can lead to *"Turing-complete"* semantics (which we have wanted to avoid) and too complex type resolution/validation mechanisms.

Reply:
At this moment we have decided don't change this; (in our Telegram dialog)